system/Database/OCI8/Forge.php

```diff
@@ -273,18 +273,22 @@ protected function _dropTable(string $table, bool $ifExists, bool $cascade)
         return $sql;
     }
 
+    /**
+     * Generates SQL to process foreign keys
+     */
     protected function _processForeignKeys(string $table): string
     {
-        $sql = '';
+        $allowActions = ['CASCADE', 'SET NULL', 'NO ACTION'];
 
-        $allowActions = [
-            'CASCADE',
-            'SET NULL',
-            'NO ACTION',
-        ];
+        $sql = '';
 
         foreach ($this->foreignKeys as $fkey) {
-            $nameIndex            = $table . '_' . implode('_', $fkey['field']) . '_fk';
+            $nameIndex = $table . '_' . implode('_', $fkey['field']) . '_fk';
+
+            if (version_compare($this->db->getVersion(), '12.2.0', '<') && strlen($nameIndex) > 30) {
+                $nameIndex = substr($nameIndex, 0, 28) . mt_rand(10, 99);
+            }
```

Comment:
This naming scheme makes it impossible to share exactly the same table schema with the DB migration.
It is not a good practice and the method of naming using random strings is not appropriate for CodeIgniter.